src/frontend/src/modals/IOModal/components/IOFieldView/components/sessionSelector/newSessionSelector.tsx

```diff
@@ -0,0 +1,211 @@
+  inspectSession: (session: string) => void;
+  updateVisibleSession: (session: string) => void;
+  selectedView?: { type: string; id: string };
+  setSelectedView: (view: { type: string; id: string } | undefined) => void;
+}) {
+  const currentFlowId = useFlowStore((state) => state.currentFlow?.id);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedSession, setEditedSession] = useState(session);
+  const { mutate: updateSessionName } = useUpdateSessionName();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setEditedSession(session);
+  }, [session]);
+
+  const handleEditClick = (e?: React.MouseEvent<HTMLDivElement>) => {
+    e?.stopPropagation();
+    setIsEditing(true);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedSession(e.target.value);
+  };
+
+  const handleConfirm = () => {
+    setIsEditing(false);
+    if (editedSession.trim() !== session) {
+      updateSessionName(
+        { old_session_id: session, new_session_id: editedSession.trim() },
+        {
+          onSuccess: () => {
+            if (isVisible) {
+              updateVisibleSession(editedSession);
+            }
+            if (
+              selectedView?.type === "Session" &&
+              selectedView?.id === session
+            ) {
+              setSelectedView({ type: "Session", id: editedSession });
+            }
+          },
+        },
+      );
+    }
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    setEditedSession(session);
+  };
+
+  const handleSelectChange = (value: string) => {
+    switch (value) {
+      case "rename":
+        handleEditClick();
+        break;
+      case "messageLogs":
+        inspectSession(session);
+        break;
+      case "delete":
+        deleteSession(session);
+        break;
+    }
+  };
+
+  return (
+    <div
+      data-testid="session-selector"
+      onClick={(e) => {
+        if (isEditing) e.stopPropagation();
+        else toggleVisibility();
+      }}
+      className={cn(
+        "file-component-accordion-div group cursor-pointer rounded-md hover:bg-muted-foreground/30",
+        isVisible ? "bg-muted-foreground/15" : "",
+      )}
+    >
+      <div className="flex w-full items-center justify-between gap-2 overflow-hidden px-2 py-1 align-middle">
+        <div className="flex min-w-0 items-center gap-2">
+          {isEditing ? (
+            <div className="flex items-center">
+              <Input
+                ref={inputRef}
+                value={editedSession}
+                onKeyDown={(e) => {
+                  if (e.key === "Enter") {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    handleConfirm();
+                  }
+                }}
+                onChange={handleInputChange}
+                onBlur={(e) => {
+                  console.log(e.relatedTarget);
+                  if (
```

Comment:
can we have a function to each interaction inside the HTML, to improve readability?